migration/src/main/java/uk/ac/bbsrc/tgac/miso/migration/destination/DefaultMigrationTarget.java

```diff
@@ -191,7 +191,7 @@ private void saveSample(Sample sample) throws IOException {
         if (detailed.getParent().getSampleClass() == null && detailed.getParent().getPreMigrationId() != null) {
           // find previously-migrated parent
           detailed.setParent((DetailedSample) serviceManager.getSampleDao().getByPreMigrationId(detailed.getParent().getPreMigrationId()));
-          if (detailed.getParent() == null) {
+          if (detailed.getParent() != null) {
```

Comment:
A change is needed though to avoid the NPE (and actually throw the IOException) though. Either store `preMigrationId` in a separate variable before lookup, and use that for the log message, or `getByPreMigrationId` into a new variable and throw the exception if null before actually changing `detailed`'s parent